app/domain/people_skills_filter.rb

```diff
@@ -25,7 +26,33 @@ def filter_by_rated
     entries
   end
 
-  def filter_by_level(entries)
-    level.nil? ? entries : entries.where('level >= ?', level)
+  def filter_by_level(entries, skills)
+    return entries if level.nil?
+    skill_ids = skills.to_s.split(',')
+    if skill_ids.length == @level.length
+      return get_filter_results(entries, skill_ids)
+    else
+      raise ArgumentError, 'Amount of Skill_ids and levels provided do not match'
+    end
+  end
+
+  def get_filter_results(entries, skill_ids)
+    person_ids = []
+    entries.where(buildstring(skill_ids)).group(:person_id).count.each_entry do |key, value|
+      person_ids.push(key) if value == skill_ids.length
+    end
+    entries.where(person_id: person_ids, skill_id: skill_ids)
+  end
+
+  def buildstring(skill_ids)
+    result = +''
+    skill_ids.each_with_index do |skill, index|
+      result.concat('(skill_id=')
+            .concat(skill.to_s)
+            .concat(' and level >=')
+            .concat(@level[index].to_s)
+            .concat(') or ')
+    end
```

Comment:
You can also try to build a string like this for the OR statement just two wishes:
1) change the method name to something that makes genuine sense and makes it readable
2) escape the params: https://guides.rubyonrails.org/active_record_querying.html#array-conditions

This code could easily allow sql injection which is really dangerous!